File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.java
class Solution {

    public int longestConsecutive(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        int result = 0;

        for (int n : nums) {
            set.add(n);
        }

        int length = 1;
        int current = 0;
        for (int n : set) {
            if (!set.contains(n -1)) {
                current = n;
                length = 1;
                while(set.contains(current + 1)) {
                    current++;
                    length++;
                }
                result = Math.max(result, length);
            }
        }
        return result;
    }
}

File: Data Structures & Algorithms/house-robber/submission-1.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        memo = {}
        n = len(nums)
        def dp(index):
            if index in memo:
                return memo[index]
            
            if index >= n:
                return 0

            memo[index] = max(nums[index] + dp(index + 2), dp(index + 1))
            return memo[index]

        return dp(0)